<launch>
  <arg name="VIEW" default="false" />

  <!-- setting -->
  <rosparam param="max_steering"> 8.37758041 </rosparam> <!-- -480[deg] ~ 480[deg] -->
  <rosparam param="mochikae_max_steering"> 8.37758041 </rosparam> <!-- 6.283185307 </rosparam> --><!-- 1.93731546971 </rosparam> --> <!-- 111[deg] -->
  <rosparam param="mochikae_min_steering"> -8.37758041 </rosparam> <!-- -6.283185307 </rosparam> --><!-- -1.93731546971 </rosparam> --> <!-- -111[deg] -->
  <rosparam param="mochikae_threshold_factor"> 0.05 </rosparam>
  <rosparam param="path_num"> 15 </rosparam> <!-- this should be odd and more than 3 -->
  <rosparam param="field_of_vision"> 80 </rosparam>
  <rosparam param="wheelbase"> 2.05 </rosparam>
  <rosparam param="tread"> 1.32 </rosparam>
  
  <!-- KdTree and Visualization Marker for obstacle detection -->
  <node name="simplified_local_planner_mochikae" pkg="drc_task_common" type="simplified_local_planner_mochikae" output="screen" >
    <remap from="points" to="/local_planner/kdtree_curve/input" />
    <remap from="steering" to="/staro_drive/hand_wheel/steering_angle" />
    <remap from="goal_dir" to="/cheat_goal_dir/ang" />
    <remap from="real_robot/stepon_gaspedal" to="/staro_drive/operation/flag/handle" />
    <remap from="real_robot/mochikae/min_max" to="/staro_drive/mochikae/min_max" />
    <remap from="curve_path/points2" to="/curve_path/points2" />
    <remap from="local_planner/steering/index_cmd" to="/hand_wheel/local_planner/input" />
    <remap from="local_planner/steering/cmd" to="/hand_wheel/local_planner/raw_cmd_angle" />
  </node>
  
  <!-- publish and Visualize steering angle -->
  <node pkg="drc_task_common" type="SteeringAngleMarker.py" name="steering_angle_marker" output="screen" >
    <remap from="cmd" to="/hand_wheel/local_planner/input" />
    <remap from="marker_array" to="/local_planner/steering/marker_array" />
  </node>
  
  <!-- switch input cloud of local planner -->
  <node name="switch_input_cloud" pkg="drc_task_common" type="switch_input_cloud" output="screen" >
    <remap from="passthrough/points" to="/passthrough/stereo/remove_xz" />
    <remap from="raw/points" to="/transform_cloud/points2" />
    <remap from="kdtree_curve/input" to="/local_planner/kdtree_curve/input" />
  </node>

  <group if="$(arg VIEW)">
    <include file="$(find drc_task_common)/launch/visualize_local_planner_viso2.launch" />
  </group>
</launch>